cranelift/codegen/src/isa/zkasm/lower/isle.rs

```diff
@@ -364,6 +364,20 @@ impl generated_code::Context for ZkAsmIsleContext<'_, '_, MInst, ZkAsmBackend> {
     fn store_op(&mut self, ty: Type) -> StoreOP {
         StoreOP::from_type(ty)
     }
+
+    fn is_global_base(&mut self, name: ExternalName) -> Option<ExternalName> {
+        match name {
+            ExternalName::User(user_name_ref) => {
+                if user_name_ref.as_u32() == 1 {
```

Comment:
Ideally this check would be more robust and less coupled with the environment logic...